rubicon_ml/client/project.py

```diff
@@ -347,6 +352,84 @@ def dataframes(self, tags=[], qtype="or", recursive=False, name=None):
 
         return self._dataframes
 
+    @failsafe
+    def archive(self, experiment_names: Optional[list[str]] = None):
+        """Archive the experiments logged to this project.
+        Parameters
+        ----------
+        experiment_names : list of str, optional
+            The rubicon.client.Experiment names to aarchive.
+
+        """
+        if len(self.experiments()) == 0:
+            raise ValueError("`project` has no logged `experiments` to archive")
+        if experiment_names is not None:
+            if not isinstance(experiment_names, list) or not all(
+                [isinstance(experiment_name, str) for experiment_name in experiment_names]
+            ):
+                raise ValueError("`experiment_names` must be `list` of type `str`")
+
+        archive_dir = os.path.join(self.repository.root_dir, self.name, "archives")
+        ts = datetime.timestamp(datetime.now())
+        archive_path = os.path.join(archive_dir, "archive-" + str(ts))
+        zip_archive_filename = str(archive_path + ".zip")
+        experiments_path = self.repository._get_experiment_metadata_root(self.name)
+
+        if experiment_names is None:
+            shutil.make_archive(archive_path, "zip", experiments_path)
+        else:
+            experiment_paths = []
+            for experiment_name in experiment_names:
+                experiment = self.experiment(name=experiment_name)
+                experiment_paths.append(
+                    self.repository._get_experiment_metadata_path(self.name, experiment.id)
+                )
+            self.repository._mkdir(archive_dir)
+            with ZipFile(zip_archive_filename, "x") as archive:
+                for file_path in experiment_paths:
+                    archive.write(file_path, os.path.basename(file_path))
+                archive.close()
+
+        if os.path.exists(zip_archive_filename):
+            print("zip archive created")
+        else:
+            print("zip archive not created")
+
+    @failsafe
+    def experiments_from_archive(self, remote_root: str, latest_only: Optional[bool] = False):
```

Comment:
we'll probably need to replace these `os` calls that interact with the filesystem with `fsspec` calls that do the same thing in order to get it working with S3